sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -659,8 +663,9 @@ def test_check_estimator():
     with raises(ValueError, match=msg):
         check_estimator(UntaggedBinaryClassifier())
 
-    # non-regression test for estimators transforming to sparse data
-    check_estimator(SparseTransformer())
+    for csr_container in CSR_CONTAINERS:
+        # non-regression test for estimators transforming to sparse data
+        check_estimator(SparseTransformer(sparse_container=csr_container))
```

Comment:
Ah, because we cannot use pytest here, now I got it.

Reply:
We still should be able to decorate this test with `pytest.mark.parametrize` at the top, but we cannot do it within the scope of the function.

(I think we could parametrize this test).

Reply:
nop because you will import `pytest` on the top of the file to do that.

Reply:
Is there something preventing us from importing `pytest`?

Reply:
Yes the CI, cf. (top of the file):

```
# We can not use pytest here, because we run
# build_tools/azure/test_pytest_soft_dependency.sh on these
# tests to make sure estimator_checks works without pytest.
```

If we import `pytest` then we will get an import error for this specific CI build.

Reply:
This shell-script uninstalls pytest, (as I found out the other day by stupidly running it). :laughing: